packages/core/src/external/fhir/hydration/resources/condition.ts

```diff
@@ -0,0 +1,209 @@
+    if (
+      code &&
+      !clinicalStatusCodeToHl7CodeMap[code] &&
+      !knownBlacklistedClinicalStatuses.includes(code)
+    ) {
+      capture.message("Unknown Condition.clinicalStatus code", {
+        level: "warning",
+        extra: {
+          unknownCode: code,
+          system: coding.system,
+          display: coding.display,
+          patientId,
+        },
+      });
+    }
+  }
+}
+
+export function buildUpdatedClinicalStatus(
+  existingStatus: CodeableConcept | undefined
+): CodeableConcept | undefined {
+  if (!existingStatus) return undefined;
+
+  const validStatusCodings = existingStatus.coding?.filter(coding => {
+    const code = coding.code?.toLowerCase()?.trim();
+    if (!code) return false;
+    return clinicalStatusCodeToHl7CodeMap[code] || knownBlacklistedClinicalStatuses.includes(code);
+  });
+
+  if (!validStatusCodings || validStatusCodings.length < 1) return undefined;
+
+  const validExistingStatus = {
+    ...existingStatus,
+    coding: validStatusCodings,
+  };
+
+  if (
+    validExistingStatus.coding?.some(
+      coding => coding.system === CONDITION_CLINICAL_STATUS_URL && coding.code
+    )
+  ) {
+    return validExistingStatus;
+  }
+
+  const hl7ClinicalStatusCoding = buildHl7ClinicalStatusCoding(validStatusCodings);
+  if (!hl7ClinicalStatusCoding) return validExistingStatus;
+
+  return {
+    ...validExistingStatus,
+    // Add the HL7 code to the top of the coding array
+    coding: [hl7ClinicalStatusCoding, ...(validExistingStatus.coding ?? [])],
+  };
+}
+
+function buildHl7ClinicalStatusCoding(codings: Coding[]): Coding | undefined {
+  for (const coding of codings) {
+    const code = coding.code?.toLowerCase()?.trim();
+    if (!code) continue;
+
+    const hl7Code = clinicalStatusCodeToHl7CodeMap[code];
+    if (hl7Code) {
+      return {
+        system: CONDITION_CLINICAL_STATUS_URL,
+        code: hl7Code,
+        display: toTitleCase(hl7Code),
+      };
+    }
+  }
+
+  return undefined;
+}
+
+function buildUpdatedCategory(condition: Condition, encounters: Encounter[]): CodeableConcept[] {
+  if (
+    condition.category?.find(c =>
+      c.coding?.find(coding => coding.system === CONDITION_CATEGORY_SYSTEM_URL)
+    )
+  ) {
+    return condition.category;
+  }
+
+  const hl7Category = buildHl7CategoryBasedOnHeuristics(condition, encounters);
+
+  // We specifically put the Hl7 category into a separate element of the category
+  // array because its semantic meaning is likely different from other systems' categories.
+  return [{ coding: [hl7Category] }, ...(condition.category ?? [])];
+}
+
+/**
+ * Decides the condition category based on the following heuristics:
+ *
+ * 1. If any Condition.code text or display contains words "history of", it's a problem-list-item.
+ * 2. If there's an ICD-10 code that starts with a Z, it's a problem-list-item.
+ * 3. If Encounter.diagnosis references the Condition, it's an encounter-diagnosis.
+ * 4. Default to problem-list-item.
```

Comment:
Id rather not include the category if we cant determine it rather than say it comes from a problem-list when it does not.